### PR TITLE
Prep patches for ignition-liveapply

### PR DESCRIPTION
**Pull request by jlebon**
Split out of https://github.com/coreos/ignition/pull/1285. See individual commits.

### CODE REVIEW

File: internal/exec/config_fetcher.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	State   *state.State
+}
+
+// RenderConfig evaluates "ignition.config.replace" and "ignition.config.append"
+// in the given config and returns the result. If "ignition.config.replace" is
+// set, the referenced and evaluted config will be returned. Otherwise, if
+// "ignition.config.append" is set, each of the referenced configs will be
+// evaluated and appended to the provided config. If neither option is set, the
+// provided config will be returned unmodified. An updated fetcher will be
+// returned with any new timeouts set.
```
Comment (bgilbert) [inline internal/exec/config_fetcher.go:46]: Might as well update this comment block for spec 3 config merging while we're here.

File: internal/exec/config_fetcher.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	f.Logger.LogReport(r)
+	if err != nil {
+		return types.Config{}, err
+	}
+
+	f.State.FetchedConfigs = append(f.State.FetchedConfigs, state.FetchedConfig{
+		Kind:       "user",
+		Source:     u.Path,
+		Referenced: true,
+	})
```
Comment (bgilbert) [inline internal/exec/config_fetcher.go:143]: This block moved within the function.  It's a valid bugfix, but let's either do that in a separate commit or at least call it out in the commit message.

File: internal/exec/stages/mount/mount.go [hunk 3]
```diff
@@ -68,7 +68,14 @@ func (stage) Name() string {
 	return name
 }
 
+func isNoOp(config types.Config) bool {
+	return len(config.Storage.Filesystems) == 0
+}
+
 func (s stage) Run(config types.Config) error {
+	if isNoOp(config) {
```
Comment (bgilbert) [inline internal/exec/stages/mount/mount.go:76]: The new check here and in umount stage is redundant, so normally I'd be inclined to omit it.  I assume this is needed for future work, though.

**Comment by jlebon** [comment]
Updated!

File: internal/exec/stages/mount/mount.go (hunk 3 above)
Comment (bgilbert) [reply internal/exec/stages/mount/mount.go:76]: Yeah, if `isNoOp` isn't going to become part of the stage API, I feel like they're slightly misleading and would prefer to drop.

Comment (jlebon) [reply internal/exec/stages/mount/mount.go]: Updated!

File: internal/exec/config_fetcher.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	Logger  *log.Logger
+	Fetcher *resource.Fetcher
+	State   *state.State
+}
+
+// RenderConfig evaluates "ignition.config.replace" and "ignition.config.merge"
+// in the given config and returns the result. If "ignition.config.replace" is
+// set, the referenced and evaluted config will be returned. Otherwise, if
+// "ignition.config.merge" is set, each of the referenced configs will be
+// evaluated and appended to the provided config. If neither option is set, the
```
Comment (bgilbert) [inline internal/exec/config_fetcher.go:44]:
```suggestion
// evaluated and merged with the provided config. If neither option is set, the
```

**Review by bgilbert — APPROVED** [review]

File: internal/exec/config_fetcher.go (hunk 4 above)
Comment (jlebon) [reply internal/exec/config_fetcher.go]: Fixed! As well as some instances in the code itself.

**Review by bgilbert — APPROVED** [review]
